include/revng/Model/CommonTypeMethods.h

```diff
@@ -0,0 +1,224 @@
+  const auto &asPrimitive() const { return *tryGetAsPrimitive(); }
+  model::PrimitiveType *tryGetAsPrimitive();
+  const model::PrimitiveType *tryGetAsPrimitive() const;
+  bool isPrimitive() const { return tryGetAsPrimitive() != nullptr; }
+
+  /// This is a specialized version of \ref asDefinition(), see
+  /// the documentation there for more details.
+  model::ArrayType &asArray() { return *tryGetAsArray(); }
+  const model::ArrayType &asArray() const { return *tryGetAsArray(); }
+  model::ArrayType *tryGetAsArray();
+  const model::ArrayType *tryGetAsArray() const;
+  bool isArray() const { return tryGetAsArray() != nullptr; }
+
+  /// This is a specialized version of \ref asDefinition(), see
+  /// the documentation there for more details.
+  model::PointerType &asPointer() { return *tryGetAsPointer(); }
+  const model::PointerType &asPointer() const { return *tryGetAsPointer(); }
+  model::PointerType *tryGetAsPointer();
+  const model::PointerType *tryGetAsPointer() const;
+  bool isPointer() const { return tryGetAsPointer() != nullptr; }
+
+  /// This helper returns true if and only if this type is is a primitive or
+  /// a pointer.
+  ///
+  /// \note it asserts on `void`.
+  bool isScalar() const;
+
+public:
+  /// Use this method to check whether a specific type is constant or not, as
+  /// opposed to the \ref IsConst() member (which should be reserved for
+  /// assigning `const`ness), since it unwraps typedefs and arrays.
+  bool isConst() const;
+
+  /// Skip all the "wrappers" around the definition.
+  ///
+  /// This is provided for some very specific cases, please take extra care when
+  /// using it. If you are hesitating whether you need this, please shop around
+  /// other functions provided here. If any of them sounds like it should do
+  /// the trick, prefer it instead.
+  ///
+  /// \returns a pointer to the definition, or a `nullptr` for primitive types
+  ///          that do not have one.
+  ///
+  /// \note in the cases when this returns `nullptr`, the type does not have
+  ///       a definition, as in, the inner-most node is a primitive.
+  model::TypeDefinition *skipToDefinition();
+  const model::TypeDefinition *skipToDefinition() const;
+
+  /// An even more specialized version of \ref skipToDefinition.
+  ///
+  /// It only differs in the sense that the found `DefinedType` is returned as
+  /// is instead of getting automatically unwrapped. Which bring inability
+  /// to call it on a `model::TypeDefinition`: it will abort.
+  model::DefinedType *skipToDefinedType();
+  const model::DefinedType *skipToDefinedType() const;
+
+public:
+  /// Returns the pointee type while unwrapping typedefs.
+  ///
+  /// \note This asserts if the type is _not_ a pointer, so only use it when
+  ///       it's guaranteed to be one.
+  ///
+  /// \note This also ignores the constness of the dropped pointer (and typedefs
+  ///       in the way)
+  model::Type &stripPointer();
+  const model::Type &stripPointer() const;
```

Comment:
Rename to `getPointee`. `stripPointer` is an action and it makes me think we're changing something, while we're not.